### ISSUE TITLE
Surface Integrals fails in parallel

### ISSUE THREAD

**Issue by arijitlaik** [underworldcode/underworld2#356]
I am not sure if this happened before, if i use surface integrals now in parallel it seems to fail on the rank where the top(or the surfaceIndexSet) in not it's domain, 
Sorry for not providing a MVE but the 1_05_StokesSinker example seems to fail in parallel as well,
FYI: just tested in v27
```python error
Traceback (most recent call last):
  File "1_05_StokesSinker.py", line 271, in <module>
An uncaught exception was encountered on processor 0.
    fn=1.0, mesh=mesh, integrationType='surface', surfaceIndexSet=top)
  File "/opt/underworld2/underworld/_stgermain.py", line 101, in __call__
    self = super(_SetupClass, cls).__call__(*args, **kwargs)
  File "/opt/underworld2/underworld/utils/_utils.py", line 102, in __init__
    raise RuntimeError("For surface integration, you must provide a 'surfaceIndexSet'.")
RuntimeError: For surface integration, you must provide a 'surfaceIndexSet'.
application called MPI_Abort(comm=0x84000004, 1) - process 0
```

**Comment by jmansour** [comment]
Yeah, this is fixed on dev, but not on v27.  I'll port the fix over to v27.  Thanks for reporting!

**Comment by arijitlaik** [comment]
@jmansour thanks, :) and Happy Holidays ;)

**Comment by arijitlaik** [comment]
@jmansour any updates on this?

**Comment by jmansour** [comment]
Hi @arijitlaik.  The fix is in!  0ec091734035311ddba2b155a9edc1e0e835d599

Can you test and then close this tix if all good?

**Comment by arijitlaik** [comment]
Thanks :) I'll test it out later in the morning and close this :)
